### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by roborourke**
This is just bringing Google Tag Manager functionality and some config integration in right now.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "name": "humanmade/analytics",
+    "description": "Analytics module for Platform",
+    "type": "package",
+    "require": {
+        "humanmade/hm-gtm": "1.1.0"
```
Comment (rmccue) [inline composer.json]: Presuming this needs adding to core too?

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    },
+    "license": "GPL-3.0",
```
Comment (rmccue) [inline composer.json]: I think this should be 2-or-later to match the other modules.

File: inc/google-tag-manager/namespace.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * @package hm-platform/analytics
+ */
+
+namespace HM\Platform\Analytics\Google_Tag_Manager;
+
+use function HM\Platform\get_config as get_platform_config;
+use HM_GTM\Plugin;
+
+function bootstrap() {
```
Comment (rmccue) [inline inc/google-tag-manager/namespace.php]: phpDoc for all of these.

File: inc/namespace.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * @package hm-platform/analytics
+ */
+
+namespace HM\Platform\Analytics;
+
+use const HM\Platform\ROOT_DIR;
+use HM\Platform\Module;
+
+function bootstrap( Module $module ) {
```
Comment (rmccue) [inline inc/namespace.php]: phpDoc here too

File: load.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+namespace HM\Platform\Workflow;
+
+use const HM\Platform\ROOT_DIR;
+use HM\Platform\Module;
+
+function bootstrap( Module $module ) {
+	$settings = $module->get_settings();
+
+	if ( $settings['google-tag-manager'] ) {
+		add_action( 'muplugins_loaded', __NAMESPACE__ . '\\load_gtm' );
```
Comment (rmccue) [inline load.php]: Any reason not to load with `plugins_loaded` instead?

File: composer.json (hunk 2 above)
Comment (roborourke) [reply composer.json]: Ah ok. I've defaulting to GPL v3 everywhere recently.

File: load.php (hunk 5 above)
Comment (roborourke) [reply load.php]: Yep, pretty common for plugins to use (and expect to be able to use) the `plugins_loaded` hook themselves so I don't think we should be loading anything from a module on that hook.

File: composer.json (hunk 1 above)
Comment (roborourke) [reply composer.json]: PR at humanmade/platform-core#13

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# Analytics
+
+Analytics is the gathering, understanding and application of data to measure, improve and make sites smarter.
+
+## Google Tag Manager
+
+There are many 3rd party providers of analytics so this module provides a simple interface to add [Google Tag Manager](https://tagmanager.google.com/) containers, the recommended way to add and manage 3rd party JavaScript on your site.
+
+Google Tag Manager provides comprehensive integration with [Google Analytics](google-tag-manager/google-analytics.md), [Google Optimise](google-tag-manager/optimise.md) and other tools in the 360 suite. Additionally there is built in support for services including ChartBeat, comScore, Parse.ly, Hotjar and [many more](https://support.google.com/tagmanager/answer/6106924).
```
Comment (joehoyle) [inline docs/README.md]:
```suggestion
Google Tag Manager provides comprehensive integration with [Google Analytics](google-tag-manager/google-analytics.md), [Google Optimise](google-tag-manager/optimise.md) and other tools in the 360 suite. Additionally there is built in support for services including [ChartBeat](https://chartbeat.com/), [comScore](https://www.comscore.com/), [Parse.ly](https://www.parse.ly/), [Hotjar](https://www.hotjar.com/) and [many more](https://support.google.com/tagmanager/answer/6106924).
```

**Pull request merged**